kizhin.evgeniy/P2

sdjklfjsdlkfjsdklfj opened the pull request.

Commit:
Initial commit for P2

Commit:
Update taylor_cosh.hpp

Added declaration of function computeTaylorCosh

Commit:
Update taylor_cosh.cpp

Added defenition of function computeTaylorCosh

Commit:
Update print_table.hpp

Added declarations of template functions printCell and printRow

Commit:
Update main.cpp

Added definition of function main that inputs interval bounds and
maximum number of operands and outputs a comparison table for both
std::cosh and Taylor series cosh from taylor_cosh.cpp

Commit:
Update print_table.cpp

Added definitions of template functions printCell and printRow

Commit:
Refactor all files

- Renamed print_table.* to printing_utils.*
- Added const modificator
- Added identations into namespaces
- Other minor changes

Commit:
Update printing_utils

Updated header guards

Commit:
Update printing_utils

Removed templates (used overloads instead)

Volgarenok commented on kizhin.evgeniy/P2/printing_utils.cpp:
```diff
@@ -0,0 +1,28 @@
+  printCell(os, argument);
+  printCell(os, errorMessage);
+  printCell(os, expectedValue);
+  os << '\n';
```
Символ конца строки обычно выводится за пределами функции вывода. Дописать его всегда можно снаружи, а вот стереть нельзя. Это не работает, скажем, для функции вывода матрицы. Но работает для функции вывода строки матрицы

Volgarenok commented on kizhin.evgeniy/P2/printing_utils.cpp:
```diff
@@ -0,0 +1,28 @@
+  os << std::setw(width) << std::left << std::setprecision(precision) << data;
+}
+
+void kizhin::printRow(std::ostream& os, double argument, const char* errorMessage,
+                      double expectedValue)
```
В CG есть пример с переносом сигнатуры. В таком виде точно неприемлемо

Volgarenok commented on kizhin.evgeniy/P2/taylor_cosh.cpp:
```diff
@@ -0,0 +1,26 @@
+  }
+  double result = 1.0;
+  double current = 1.0;
+  double error = absoluteError + 1;
```
Не очень понял. Поясните

sdjklfjsdlkfjsdklfj replied:
result и current инициализируется единицей, тк это единица первое слагаемое в ряде Тейлора для функции гипреболического косинуса (таким образом в цикле мы начинаем вычисление со второго слагаемого). 

погрешность инициализируется таким образом, тк предполагалось, что в цикле может не быть интераций, а значит погрешность не обновится, в таком случае логичнее было бы инициализировать ее единицей (исправил в последнем коммите)

Commit:
Update printing_utils and taylor_cosh.cpp

- Updated indentations
- Updated variable initializations into taylor_cosh
- Minor changes

Commit:
Update taylor_cosh.cpp

- Minor changes

Commit:
Update taylor_cosh.cpp

I'm so sorry T-T

Commit:
Merge remote-tracking branch 'upstream/master' into kizhin.evgeniy/P2
